src/funcs.c

```diff
@@ -1335,7 +1362,11 @@ void game_properties(void)
 	if (game_tooltypes)
 		memset(&game_tooltypes[0], 0, 1024 * sizeof(char));
 	else
+#if defined(__amigaos4__)
+		game_tooltypes = AllocVecTags(sizeof(char *) * 1024, AVT_ClearWithValue,0, TAG_DONE);
+#else
 		game_tooltypes = AllocMem(1024 * sizeof(char), MEMF_CLEAR);
+#endif
 
```

Comment:
In many places, you are doing the same if and the same code on how to reserve memory. It would be better if this could be transferred into a single method that does the check and covers all possible situations. This way, the maintenance would be  easier.